compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/InheritedDefaultMethodsOnClassesLowering.kt

```diff
@@ -180,7 +212,28 @@ private class InterfaceDefaultCallsLowering(val context: JvmBackendContext) : Ir
         // gets redirected to call itself.
         if (redirectTarget == currentFunction?.irElement) return super.visitCall(expression)
 
-        val newCall = irCall(expression, redirectTarget, receiversAsArguments = true)
+        val newCall = IrCallImpl(
+            expression.startOffset,
+            expression.endOffset,
+            expression.type,
+            redirectTarget.symbol,
+            typeArgumentsCount = redirectTarget.typeParameters.size,
+            valueArgumentsCount = redirectTarget.valueParameters.size,
+            origin = expression.origin
+        ).apply {
+            copyValueArgumentsFrom(
+                expression,
+                expression.symbol.owner,
+                this.symbol.owner,
+                receiversAsArguments = true,
+                argumentsAsReceivers = false
+            )
+            //TODO: See Note [Interface Parameters]
+            var offset = 0
+            expression.symbol.owner.parentAsClass.typeParameters.forEach {
+                putTypeArgument(offset++, context.irBuiltIns.anyNType)
```

Comment:
Technically, `Any?` might not be a valid instantiation for a given type parameter, since the parameter could have bounds. Could you instantiate it with `it.defaultType` instead?

Reply:
You are right, I actually think two of these cases can be done correctly! Rewriting, pushing a revision.